PlastomeRegionBurstAndAlign.py

```diff
@@ -340,165 +322,154 @@ def main(args):
         if select_mode == 'int':
             ExtractAndCollect(main_d_nucl).do_INT(main_d_intron2, rec, log)
             main_d_nucl.update(main_d_intron2)
-        # Check if main dictionary empty
+
         if not main_d_nucl.items():
-            log.critical("No items in main dictionary" % out_dir)
+            log.critical("No items in main dictionary")
             raise Exception()
-    # --------------------------------------------------------------------------#
-    # REMOVE DUPLICATE ANNOTATIONS
-    # Note: Not sure why I have to run this removal twice, but not all
-    #       duplicates are removed first time around.
-    action = "removing duplicate annotations"
-    log.info("%s" % action)
-    ###
-    remove_duplicates(main_d_nucl)
-    remove_duplicates(main_d_nucl)
-    if select_mode == 'cds':
-        remove_duplicates(main_d_prot)
-        remove_duplicates(main_d_prot)
-    # --------------------------------------------------------------------------#
-    # REMOVE ANNOTATIONS THAT OCCUR IN FEWER THAN X TAXA
-    action = ("removing annotations that occur in fewer than %s taxa" % tax_cutoff)
-    log.info("%s" % action)
-    ###
+
+    return main_d_nucl, main_d_prot
+
+def remove_annotations_below_taxa_cutoff(main_d_nucl, main_d_prot, tax_cutoff):
     for k, v in main_d_nucl.items():
         if len(v) < tax_cutoff:
             del main_d_nucl[k]
-            if select_mode == 'cds':
```

Comment:
Can you please keep the line ```if select_mode == 'cds':``` and just comment it out. Probably your new line is just fine, but there may be incompatibility issues if just doing ```if main_d_prot```.

           # if select_mode == 'cds':